packages/kernel-6.1/latest-kernel-full-config.sh

```diff
@@ -0,0 +1,146 @@
+EOF
+}
+
+usage_error() {
+    >&2 usage
+    bail "$1"
+}
+
+check_dependencies() {
+    hash rpm2cpio || usage_error "DEPENDENCY ERROR: Please install rpm2cpio somewhere in your PATH"
+    hash cpio || usage_error "DEPENDENCY ERROR: Please install cpio somewhere in your PATH"
+    hash tar || usage_error "DEPENDENCY ERROR: Please install tar somewhere in your PATH"
+    hash docker || usage_error "DEPENDENCY ERROR: Please install docker somewhere in your PATH"
+}
+
+# expect $pwd to be packages/kernel-${kver}
+merge_kernel_configs() {
+    rpm_file=$1
+    kernel_path=$PWD
+
+    version="$(rpm --query --nosignature --queryformat '%{VERSION}' "${rpm_file}")"
+    readarray -t br_patches < <(find "${kernel_path}" -name "*.patch")
+
+    tmpdir=$(mktemp -d)
+    cp "$rpm_file" "$tmpdir/"
+    pushd "$tmpdir" || bail "Could not move around"
+
+    rpm2cpio "${rpm_file}" | cpio -iu {,./}linux-"${version}".tar {,.}config-x86_64 {,.}config-aarch64 {,./}"*.patch" {,./}kernel.spec
+    tar -xof linux-"${version}".tar; rm linux-"${version}".tar
+
+    # Find patch ordering based upstream SRPM and apply in that order
+    readarray -t patches < <(grep -P "^Patch\d+" kernel.spec | sort -n -k1.6 | grep -oP "^Patch\d+: \K.*\.patch$" kernel.spec)
+
+    pushd "linux-${version}" || bail "Could not move into linux-${version}"
+
+    # Patches from the upstream
+    for patch in "${patches[@]}"; do
+        patch -p1 <"../$patch"
+    done
+
+    # Patches from bottlerocket
+    for patch in "${br_patches[@]}"; do
+        patch -p1 <"$patch"
+    done
+
+    popd || bail "Could not move around - 'popd' failed. Lets stop before we break anything further."
+
+    for arch in "x86_64" "aarch64"; do
+
+        # mount files from the rpm and start the sdk docker container
+        br_cfg="${kernel_path}/config-bottlerocket"
+        microcode_cfg="${kernel_path}/../microcode/config-microcode"
+        al_cfg="${PWD}/config-${arch}"
+        linux_src="${PWD}/linux-${version}"
+
+        pushd "linux-${version}" || bail "Could not move into linux-${version}"
+
+        if [ "${arch}" = "aarch64" ]; then
+            karch="arm64"
+            script_args=("../config-${arch}" "../config-bottlerocket")
+        elif [ "${arch}" = "x86_64" ]; then
+            karch="x86"
+            script_args=("../config-${arch}" "../config-microcode" "../config-bottlerocket")
+        fi
+
+        docker run --rm \
+            -v $br_cfg:/config-bottlerocket \
+            -v $microcode_cfg:/config-microcode \
+            -v $al_cfg:/config-${arch} \
+            -v $linux_src:/linux-${version} \
+            -e ARCH=${karch} \
+            -e CROSS_COMPILE=/usr/bin/${arch}-bottlerocket-linux-gnu- \
+            -e KCONFIG_CONFIG=bottlerocket_${arch}_defconfig \
+            -w /linux-${version} \
+            --name ${arch}-config-merger public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.60.0 \
+            ./scripts/kconfig/merge_config.sh \
+            "${script_args[@]}"
```

Comment:
shellcheck has some complaints about this
```suggestion
        docker run --rm \
            -v "${br_cfg}":/config-bottlerocket \
            -v "${microcode_cfg}":/config-microcode \
            -v "${al_cfg}":"/config-${arch}" \
            -v "${linux_src}":"/linux-${version}" \
            -e ARCH="${karch}" \
            -e CROSS_COMPILE="/usr/bin/${arch}-bottlerocket-linux-gnu-" \
            -e KCONFIG_CONFIG="bottlerocket_${arch}_defconfig" \
            -w "/linux-${version}" \
            --name "${arch}-config-merger" public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.60.0 \
            ./scripts/kconfig/merge_config.sh \
            "${script_args[@]}"
```

Reply:
done